core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
+	httpClient HttpClient
+}
+
+type clientPlantUML struct {
+	options optionsPlantUMLClient
+
+	baseURL string
+}
+
+func (c *clientPlantUML) Do(v DiagramGraph) (ResponseDiagram, error) {
+	code, err := diagramGraph2plantUMLCode(v)
+	if err != nil {
+		return nil, err
+	}
+	p, err := code2Path(code)
+	if err != nil {
+		return nil, err
+	}
+
+	diagram, err := c.requestHandler("svg/" + p)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseC4Diagram{SVG: *(*string)(unsafe.Pointer(&diagram))}, nil
+}
+
+func (c *clientPlantUML) requestHandler(p string) ([]byte, error) {
+	req, _ := http.NewRequest("GET", c.baseURL+p, nil)
```

Comment:
**suggestions**:

1. Replace `"GET"` by the `http.MethodGet` equivalent.
2. If decide to support `context.Context`, use `http.NewRequestWithContext()`

Reply:
Thanks for suggestions. 
1. A good one. For context, "GET" is the method which is an abstraction on its own existed even before Go and http package :) WDYT about the value of such alias?
2. Good catch! I implement it.

Reply:
Implemented